src/machine/machined-varlink.c

```diff
@@ -489,6 +491,50 @@ static int vl_method_get(sd_varlink *link, sd_json_variant *parameters, sd_varli
         return lookup_machine_by_name(link, m, mn);
 }
 
+static int vl_method_get_by_pid(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+
+        int r;
+        pid_t pid = 0;
+        assert_cc(sizeof(pid_t) == sizeof(uint32_t));
+        assert(parameters);
+
+        const sd_json_dispatch_field dispatch_table[] = {
+                { "pid", SD_JSON_VARIANT_UNSIGNED, sd_json_dispatch_uint32, PTR_TO_SIZE(&pid), SD_JSON_MANDATORY },
```

Comment:
Maybe, better to introduce `sd_json_dispatch_pid()` ?

Reply:
I thought about it. What stopped me from introducing `sd_json_dispatch_pid()` is that `pid_t` is a signed type. At the same time, `GetByPID` does not accept negative values for pid. Since this is the only case so far I don't know which way is better. So, I kept like this.